### PR TITLE
add device-specific metadata fetching

### PR DESCRIPTION
**Pull request by eugenosm**
closes  https://github.com/WoTTsecurity/api/issues/273

(agent side)

### CODE REVIEW

File: agent/__init__.py [hunk 1]
```diff
@@ -490,6 +491,43 @@ def setup_endpoints(dev, debug):
...
+            if debug:
+                print("[RECEIVED] Fetch device metadata: code {}".format(dev_md_req.status_code))
+                print("[RECEIVED] Fetch device metadata: {}".format(dev_md_req.content))
+            return
+        metadata = dev_md_req.json()
+
+        print('metadata retrieved.')
+
+        if os.path.exists(SECRET_DEV_METADATA_PATH) and not os.path.isfile(SECRET_DEV_METADATA_PATH):
+            print("Error: The filesystem object '{}' is not a file. Looks like a break-in attempt.".printf(
```
Comment (a-martynovich) [inline agent/__init__.py]: `printf`?

Comment (eugenosm) [reply agent/__init__.py]: habit from c++/go

Comment (eugenosm) [reply agent/__init__.py]: fixed

**Comment by vpetersson** [comment]
@eugenosm can this be merged before the API changes?

**Comment by eugenosm** [comment]
i'm not sure. it will try to fetch metadata each 15 min in daemon mode.

**Comment by vpetersson** [comment]
> i'm not sure. it will try to fetch metadata each 15 min in daemon mode.

But does that mean it will crash if it cannot get the metadata? If so, i need to hold this.

**Comment by a-martynovich** [comment]
Daemon never crashes even if its children crash

**Pull request merged**